packages/n4s/src/lib/transformResult/index.js

```diff
@@ -0,0 +1,62 @@
+export function isValidResult(result) {
+  return !!(
+    typeof result === 'boolean' ||
+    (result && typeof result.pass === 'boolean')
+  );
+}
+
+// for easier testing and mocking
+export function getDefaultResult(interfaceName, value, rule) {
+  return {
+    message: formatResultMessage(
+      interfaceName,
+      rule,
+      `invalid ${typeof value} value`
+    ),
+  };
+}
+
+export function formatResultMessage(interfaceName, rule, msg) {
+  return `[${interfaceName}]/${rule.name} ${msg}`;
+}
+
+/**
+ * Transform the result of a rule into a standard format
+ * @param {string} interfaceName to be used in the messages
+ * @param {*} result of the rule
+ * @param {Object} options
+ * @param {function} options.rule
+ * @param {*} options.value
+ * @returns {Object} result
+ * @returns {string} result.message
+ * @returns {boolean} result.pass indicates if the test passes or not
+ */
+export function transformResult(interfaceName, result, { rule, value }) {
+  const defaultResult = getDefaultResult(interfaceName, value, rule);
+
+  if (!isValidResult(result)) {
```

Comment:
Ok, that double negation was because ìsValidResult(null)` was returning `null`. I was not intending to triple negate, just a shortcut to ensure that it was returning a boolean.
-------------------------
I see the benefits of the `validateResult`. I'll be changing that